src/lib/FeeEscalatorLib.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {ISignatureTransfer} from "permit2/src/interfaces/ISignatureTransfer.sol";
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {Input} from "../base/ReactorStructs.sol";
+import {FeeEscalator, Input} from "../base/ReactorStructs.sol";
+
+library FeeEscalatorLib {
+    using FixedPointMathLib for uint256;
+
+    string public constant FEE_ESCALATOR_TYPESTRING =
+        "FeeEscalator(address token,uint256 startAmount,uint256 maxAmount,uint256 startTime,uint256 endTime)";
+    bytes32 internal constant FEE_ESCALATOR_TYPEHASH =
+        keccak256("FeeEscalator(address token,uint256 startAmount,uint256 maxAmount,uint256 startTime,uint256 endTime)");
+
+    /// @notice thrown if the escalation direction is incorrect
+    error InvalidAmounts();
+
+    /// @notice calculates an amount on a linear curve over time from startTime to endTime
+    /// @dev handles only increasing amounts from startAmount to endAmount
+    /// @param startAmount The amount of tokens at startTime
+    /// @param endAmount The amount of tokens at endTime
+    /// @param startTime The time to start escalating linearly
+    /// @param endTime The time to stop escalating linearly
+    function decay(uint256 startAmount, uint256 endAmount, uint256 startTime, uint256 endTime)
+        internal
+        view
+        returns (uint256 resolvedAmount)
+    {
+        if (startAmount > endAmount) {
```

Comment:
also need to check to ensure endTime > starttime?

Reply:
else youll underflow on line 40

Reply:
ah good catch